plugins/draw-tools.js

```diff
@@ -658,11 +675,95 @@ window.plugin.drawTools.boot = function() {
+};
+
+
+window.plugin.drawTools.mpe.initMPE = function(){
+  // Not launch the code if the MPE plugin there isn't.
+  if(!window.plugin.mpe){ return; }
+
+  // The MPE function to set a MultiProjects type
+  window.plugin.mpe.setMultiProjects({
+    namespace: 'drawTools',
+    title: 'Draw Tools Layer',
+    // Font awesome css class
+    fa: 'fa-pencil',
+    // Function to change a localstorage key
+    func_setKey: function(newKey){
+      window.plugin.drawTools.KEY_STORAGE = newKey;
+    },
+    // Native value of localstorage key
+    defaultKey: 'plugin-draw-tools-layer',
+    // This function is run before the localstorage key change
+    func_pre: function(){},
+    // This function is run after the localstorage key change
+    func_post: function(){
+      window.plugin.drawTools.drawnItems.clearLayers();
+      window.plugin.drawTools.load();
+      console.log('DRAWTOOLS: reset all drawn items (func_post)');
+
+      if(window.plugin.crossLinks !== undefined){
+        if(window.overlayStatus['Cross Links'] === true){
+          window.plugin.crossLinks.checkAllLinks();
+
+          if(window.plugin.destroyedLinks !== undefined){
+            if(window.overlayStatus['Destroyed Links Simulator'] === true){
+              //                                    window.plugin.destroyedLinks.cross.restoreCrossAll();
+              window.plugin.destroyedLinks.cross.removeCrossAll();
+            }
+          }
+        }
+      }
+
+      // Code to:
+      // hide/remove elements from DOM, layers, variables, etc...
+      // load data from window.localStorage[window.plugin.myPlugin.KEY_STORAGE]
+      // show/add/draw elements in the DOM, layers, variables, etc...
+    }
+  });
+}
+/* // not used
+window.plugin.drawTools.mpe.setupCSS = function(){
+    $("<style>").prop("type", "text/css").html(''
+    ).appendTo("head");
+}
+// */
+window.plugin.drawTools.setupCSS = function(){
+  $("<style>").prop("type", "text/css").html(''
+  +'.leaflet-bar{box-shadow:0 1px 5px rgba(0,0,0,.65);}'
+  +'.leaflet-draw .leaflet-draw-section .leaflet-bar{box-shadow:none;}'
+  +'.leaflet-draw{box-shadow:0 1px 5px rgba(0,0,0,.65);border-radius:4px;}'
+
+  +'.leaflet-draw .leaflet-draw-section .leaflet-bar.leaflet-draw-toolbar-top a:last-child{border-bottom:2px solid #999;}'
+  +'.leaflet-draw .leaflet-draw-section .leaflet-bar a{border-radius:0;}'
+
+  +'.leaflet-draw .leaflet-draw-section .leaflet-bar{border-radius:0 0 4px 4px;overflow:hidden;margin-top:0;}'
+  +'.leaflet-draw .leaflet-draw-section .leaflet-bar.leaflet-draw-toolbar-top{border-radius:4px 4px 0 0;}'
+).appendTo("head");
+}
+
+
 function setup () {
-  loadExternals();
-  window.plugin.drawTools.boot();
+  loadExternals();                              // initialize leaflet
+  window.plugin.drawTools.boot();               // initialize drawtools
+  window.pluginCreateHook('pluginDrawTools');
+  window.plugin.drawTools.setupCSS();           // create additional CSS elements
```

Comment:
If this function is performed not only when MPE is active but always, maybe you should transfer the CSS code to a separate CSS file?

Reply:
Need to check, where this code is needed. Might be slipped in from upcomming features in DrawToolsPlus.
If kept it should move to the other CSS code in DT.

Reply:
The CSS changes are used to remove the GAP between drawitems and drawedit to safe space.
Maybe reintroduce at a different time.
line 759: setupCSS () removed.